### PR TITLE
dump_syms: init at unstable-2022-05-05; buildMozillaMach: create buildsymbols

### PR DESCRIPTION
**Pull request by mweinelt**
Thie zip bundle in this output is used in the dumps sent by the
crashreporter. For this to happen we need to upload this zip file to
https://symbols.mozilla.org, which is a separate effort.

Fails for 91.9.0esr https://gist.github.com/mweinelt/e50ac65ec6c21ea06778947d2e228901

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by S-NA** [comment]
Is using the most recent commit of dump_syms instead of the v0.0.7 tag reasonable? v0.0.7 uses symbolic 8.0, but it seems the issue that ESR is stuck on has already been fixed in newer versions of symbolic. Ref: https://github.com/getsentry/symbolic/pull/326

**Comment by mweinelt** [comment]
Bingo, that made it work.

> firefox-91.9.0.en-US.linux-x86_64.crashreporter-symbols.zip

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -408,7 +419,13 @@ buildStdenv.mkDerivation ({
   # tests were disabled in configureFlags
   doCheck = false;
 
+  # Generate build symbols once after the final build
+  # https://firefox-source-docs.mozilla.org/crash-reporting/uploading_symbol.html
   preInstall = ''
+    ./mach buildsymbols
+    mkdir -p $symbols/
+    cp mozobj/dist/*.crashreporter-symbols.zip $symbols/
```
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/common.nix:427]: I don't particularly like putting it into `preInstall` but I want the build to create the symbols only after the second build phase in PGO builds.

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 2]
```diff
@@ -168,6 +169,11 @@ buildStdenv.mkDerivation ({
 
   inherit src unpackPhase meta;
 
+  outputs = [
+    "out"
+    "symbols"
```
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/common.nix:174]: My thought here was `symbols` as in symbols.mozilla.org?

**Comment by mweinelt** [comment]
> thunderbird-91.8.1.en-US.linux-x86_64.crashreporter-symbols.zip

Also works for thunderbird.

**Pull request merged**